components/admin_console/system_users/list/index.js

```diff
@@ -10,8 +10,11 @@ import SystemUsersList from './system_users_list.jsx';
 import {getUsers} from './selectors.jsx';
 
 function mapStateToProps(state, ownProps) {
+    const users = getUsers(state, ownProps.loading, ownProps.teamId, ownProps.term, ownProps.filter).filter((user) => {
```

Comment:
Would be nice to have this encapsulated in a selector, to cache it. If not, we are looping over the users every single time.

Reply:
Thanks @jespino I moved the logic to a selector.